components/usage/pkg/apiv1/billing.go

```diff
@@ -288,6 +321,42 @@ func (s *BillingService) CreateStripeSubscription(ctx context.Context, req *v1.C
 	}, nil
 }
 
+func (s *BillingService) UpdateCustomerSubscriptionsTaxState(ctx context.Context, req *v1.UpdateCustomerSubscriptionsTaxStateRequest) (*v1.UpdateCustomerSubscriptionsTaxStateResponse, error) {
+	if req.GetCustomerId() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "Missing CustomerID")
+	}
+
+	stripeCustomer, err := s.stripeClient.GetCustomer(ctx, req.GetCustomerId())
+	if err != nil {
+		log.WithError(err).Error("Failed to retrieve customer from Stripe.")
+		return nil, status.Errorf(codes.NotFound, "Failed to get customer with ID %s: %s", req.GetCustomerId(), err.Error())
+	}
+
+	var isAutomaticTaxSupported bool
+	if stripeCustomer.Tax != nil {
+		isAutomaticTaxSupported = stripeCustomer.Tax.AutomaticTax == stripe_api.CustomerTaxAutomaticTaxSupported
+
+		_, err = db.UpdateStripeCustomerInvalidBillingAddress(ctx, s.conn, stripeCustomer.ID, stripeCustomer.Tax.AutomaticTax == stripe_api.CustomerTaxAutomaticTaxUnrecognizedLocation)
+		if err != nil {
+			log.WithError(err).Error("Failed to update stripe customer from DB")
+		}
+	}
+	if !isAutomaticTaxSupported {
+		log.Warnf("Automatic Stripe tax is not supported for customer %s", stripeCustomer.ID)
+	}
+
+	for _, subscription := range stripeCustomer.Subscriptions.Data {
```

Comment:
Uh, that's cool!

IMO it would make sense to `log.Info` every time we update a subscription like this, so we better understand what's going on. :+1: